samples/access_token_example/test/access_token.test.js

```diff
@@ -0,0 +1,319 @@
+  const scope = 'https://www.googleapis.com/auth/devstorage.full_control';
+  // generate access token
+  const access_token = await generateGcpToken(
+    region,
+    projectId,
+    registryName,
+    deviceId,
+    scope,
+    'RS256',
+    rsaPrivateKey
+  );
+  const headers = {authorization: `Bearer ${access_token}`};
+  // Create GCS bucket
+  const createGcsPayload = {
+    name: bucketName,
+    location: region,
+    storageClass: 'STANDARD',
+    iamConfiguration: {
+      uniformBucketLevelAccess: {enabled: true},
+    },
+  };
+
+  const createGcsRequestUrl = `https://storage.googleapis.com/storage/v1/b?project=${projectId}`;
+  const createGcsOptions = {
+    url: createGcsRequestUrl,
+    method: 'POST',
+    headers: headers,
+    data: Buffer.from(JSON.stringify(createGcsPayload)),
+    'content-type': 'application/json',
+    'cache-control': 'no-cache',
+  };
+  const createResponse = await request(createGcsOptions);
+  assert.strictEqual(createResponse.status, 200);
+
+  // Upload Data to GCS bucket
+  const dataName = 'testFILE';
+  const binaryData = readFileSync('../../../resources/logo.png');
+  const uploadGcsRequestUrl = `https://storage.googleapis.com/upload/storage/v1/b/${bucketName}/o?uploadType=media&name=${dataName}`;
+  const uploadGcsOptions = {
+    url: uploadGcsRequestUrl,
+    method: 'POST',
+    headers: headers,
+    data: binaryData,
+    'content-type': 'image/png',
+    'cache-control': 'no-cache',
+  };
+  const uploadResponse = await request(uploadGcsOptions);
+  assert.strictEqual(uploadResponse.status, 200);
+
+  // Download Data from GCS bucket
+  const downloadGcsRequestUrl = `https://storage.googleapis.com/storage/v1/b/${bucketName}/o/${dataName}?alt=media`;
+  const downloadGcsOptions = {
+    url: downloadGcsRequestUrl,
+    method: 'GET',
+    headers: headers,
+    'cache-control': 'no-cache',
+  };
+  const downloadResponse = await request(downloadGcsOptions);
+  assert.strictEqual(downloadResponse.status, 200);
+
+  // Delete Data from GCS Bucket.
+  const deleteDataGcsRequestUrl = `https://storage.googleapis.com/storage/v1/b/${bucketName}/o/${dataName}`;
+  const deleteDataGcsOptions = {
+    url: deleteDataGcsRequestUrl,
+    method: 'DELETE',
+    headers: headers,
+    'cache-control': 'no-cache',
+  };
+  const deleteDataResponse = await request(deleteDataGcsOptions);
+  assert.strictEqual(deleteDataResponse.status, 200);
+
+  // Delete GCS bucket
+  const deleteGcsRequestUrl = `https://storage.googleapis.com/storage/v1/b/${createResponse.data.name}`;
+  const deleteGcsOptions = {
+    url: deleteGcsRequestUrl,
+    method: 'DELETE',
+    headers: headers,
+    'cache-control': 'no-cache',
+  };
+  const deleteResp = await request(deleteGcsOptions);
+  assert.strictEqual(deleteResp.status, 200);
+});
+
+it('Generate gcp access token, use gcp access token to create pubsub topic, push message to pubsub', async () => {
```

Comment:
please move the content from this test into access_token.js file with a new region tag `iot_access_token_pubsub` as a "feature" and add test to that.